#include "mainwindow.h"
#include "ui_mainwindow.h"
#include "canvas.h"
#include <QColorDialog>

MainWindow::MainWindow(QWidget *parent)
    : QMainWindow(parent)
    , ui(new Ui::MainWindow)
{
    ui->setupUi(this);

    // ðŸŽ¨ Ð’Ñ‹Ð±Ð¾Ñ€ Ñ†Ð²ÐµÑ‚Ð°
    connect(ui->btnColor, &QPushButton::clicked, this, [this]() {
        QColor color = QColorDialog::getColor(Qt::black, this, "Ð’Ñ‹Ð±ÐµÑ€Ð¸Ñ‚Ðµ Ñ†Ð²ÐµÑ‚");
        if (color.isValid()) {
            ui->canvas->setPenColor(color);
            ui->canvas->setDrawMode(); // ðŸ”¹ Ð°Ð²Ñ‚Ð¾Ð¼Ð°Ñ‚Ð¸Ñ‡ÐµÑÐºÐ¸Ð¹ Ð²Ñ‹Ñ…Ð¾Ð´ Ð¸Ð· Ñ€ÐµÐ¶Ð¸Ð¼Ð° Ð»Ð°ÑÑ‚Ð¸ÐºÐ°
        }
    });

    // ðŸ§½ ÐžÑ‡Ð¸ÑÑ‚ÐºÐ°
    connect(ui->btnClear, &QPushButton::clicked, this, [this]() {
        ui->canvas->clear();
    });

    // âœï¸ ÐšÐ°Ñ€Ð°Ð½Ð´Ð°Ñˆ
    connect(ui->btnPen, &QPushButton::clicked, this, [this]() {
        ui->canvas->setDrawMode();
    });

    // ðŸ©¹ Ð›Ð°ÑÑ‚Ð¸Ðº
    connect(ui->btnEraser, &QPushButton::clicked, this, [this]() {
        ui->canvas->setEraserMode(true);
    });
}

MainWindow::~MainWindow()
{
    delete ui;
}
